circuit-lib/circuit-lib.circom/benchmarks/light/transaction_light.ts

```diff
@@ -0,0 +1,169 @@
+  const poseidon = await buildPoseidonOpt();
+
+  const seed32 = bs58.encode(new Uint8Array(32).fill(1));
+  const account = new Account({ poseidon: poseidon, seed: seed32 });
+  await account.getEddsaPublicKey();
+
+  const depositAmount = 20_000;
+  const depositFeeAmount = 10_000;
+
+  const deposit_utxo1 = new Utxo({
+    poseidon: poseidon,
+    assets: [FEE_ASSET, MINT],
+    amounts: [new BN(depositFeeAmount), new BN(depositAmount)],
+    publicKey: account.pubkey,
+    assetLookupTable: lightProvider.lookUpTables.assetLookupTable,
+    verifierProgramLookupTable:
+      lightProvider.lookUpTables.verifierProgramLookupTable,
+  });
+
+  const mockPubkey: PublicKey = SolanaKeypair.generate().publicKey;
+
+  let txParams = new TransactionParameters({
+    outputUtxos: [deposit_utxo1],
+    eventMerkleTreePubkey: mockPubkey,
+    transactionMerkleTreePubkey: mockPubkey,
+    senderSpl: mockPubkey,
+    senderSol: lightProvider.wallet.publicKey,
+    action: Action.SHIELD,
+    poseidon,
+    verifierIdl: app ? IDL_VERIFIER_PROGRAM_TWO : IDL_VERIFIER_PROGRAM_ZERO,
+    account,
+  });
+
+  lightProvider.solMerkleTree!.merkleTree = new MerkleTree(
+    tree_height,
+    poseidon,
+    [
+      deposit_utxo1.getCommitment(poseidon),
+      poseidon.F.toString(poseidon(["123124"])),
+    ],
+  );
+
+  let txInputs: TransactionInputs = {
+    ...(await lightProvider.getRootIndex()),
+    solMerkleTree: lightProvider.solMerkleTree!,
+    params: txParams,
+  };
+
+  if (app) {
+    txInputs.appParams = {
+      mock: "123",
+      verifierIdl: IDL_VERIFIER_PROGRAM_ZERO,
+    };
+  }
+
+  const tx = new Transaction(txInputs);
+  await tx.compile(lightProvider.poseidon, account);
+
+  addPrivateKey(tx, account);
+  if (app) delete tx.proofInput.inPublicKey;
+
+  return tx.proofInput;
+}
+
+async function mark_transaction_account(tree_height: number, app = false) {
+  let wasm_path: string;
+  let zkey_path: string;
+  const path_modifier = app ? "App4" : "Masp2";
+
+  wasm_path = `./artifacts/transaction${path_modifier}Main${tree_height}/transaction${path_modifier}Main${tree_height}.wasm`;
+  zkey_path = `./artifacts/transaction${path_modifier}Main${tree_height}/transaction${path_modifier}Main${tree_height}.zkey`;
+
+  const proofInputs = await generateProofInputs(tree_height, app);
+  const res = await snarkjs.groth16.fullProve(
+    stringifyBigInts(proofInputs),
+    wasm_path,
+    zkey_path,
+  );
+
+  return res;
+}
+
+// skip unwanted logs
+console.log = (function (originalLog) {
+  const skipPatterns = [
+    "Provider is not defined.",
+    "Unable to fetch rootIndex. Setting root index to 0 as a default value.",
+    // Add more patterns if needed
+  ];
+
+  return function (...args) {
+    const message = args.join(" ");
+
+    if (!skipPatterns.some((pattern) => message.includes(pattern))) {
+      originalLog.apply(console, args);
+    }
+  };
+})(console.log);
+
+const iterations = 100;
```

Comment:
you could check whether circuits are built here.
If not create circuit files and build the circuits here since it doesn't count into benchtime correct?

Reply:
We should be able to execute benchmarks including building circuits in one command :)

Reply:
done!